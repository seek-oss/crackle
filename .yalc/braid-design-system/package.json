{
  "name": "braid-design-system",
  "version": "30.4.2-crackle-1",
  "description": "Themeable design system for the SEEK Group",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "files": [
    "/css",
    "/dist",
    "/reset",
    "/themes"
  ],
  "sideEffects": [
    "lib/themes/**/*",
    "lib/css/atoms/**/*",
    "lib/css/reset/**/*",
    "themes/**/*",
    "reset/**/*",
    "*.css.js"
  ],
  "repository": {
    "type": "git",
    "directory": "packages/braid-design-system",
    "url": "https://github.com/seek-oss/braid-design-system.git"
  },
  "author": "SEEK",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/seek-oss/braid-design-system/issues"
  },
  "homepage": "https://github.com/seek-oss/braid-design-system#readme",
  "dependencies": {
    "@capsizecss/core": "^3.0.0",
    "@capsizecss/vanilla-extract": "^1.0.0",
    "@vanilla-extract/css": "^1.0.0",
    "@vanilla-extract/css-utils": "^0.1.1",
    "@vanilla-extract/sprinkles": "^0.4.3",
    "assert": "^2.0.0",
    "autosuggest-highlight": "^3.1.1",
    "clsx": "^1.1.1",
    "dedent": "0.7.0",
    "gradient-parser": "^1.0.0",
    "is-mobile": "^2.2.2",
    "lodash": "^4.17.21",
    "polished": "^4.1.0",
    "react-focus-lock": "^2.5.0",
    "react-keyed-flatten-children": "^1.3.0",
    "react-popper-tooltip": "^4.0.1",
    "react-remove-scroll": "^2.4.1",
    "uuid": "^8.3.2"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17",
    "sku": ">=10.13.1"
  },
  "yalcSig": "d1bc60d35437df4f1049f52e98e5c018"
}
