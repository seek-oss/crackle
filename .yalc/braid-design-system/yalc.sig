d1bc60d35437df4f1049f52e98e5c018