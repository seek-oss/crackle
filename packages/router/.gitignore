# managed by crackle
/dist
/routes
/server
# end managed by crackle
