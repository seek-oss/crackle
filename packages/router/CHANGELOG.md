# @crackle/router

## 0.2.0

### Minor Changes

- [#48](https://github.com/seek-oss/crackle/pull/48) [`87605cf`](https://github.com/seek-oss/crackle/commit/87605cf6b6fa0deba5d1abd0dbc4e6f7318f5845) Thanks [@mrm007](https://github.com/mrm007)! - Update entrypoints

## 0.1.2

### Patch Changes

- [#6](https://github.com/seek-oss/crackle/pull/6) [`686ba96`](https://github.com/seek-oss/crackle/commit/686ba96bb1651308345786ec4c554014331356b7) Thanks [@benjervis](https://github.com/benjervis)! - `createRouteData` now accepts a static object.

  This is useful if the route data is not context dependent.

## 0.1.1

### Patch Changes

- 8e80af5: Add routes entry

## 0.1.0

### Minor Changes

- 1d3f7f9: Add strongly typed links
