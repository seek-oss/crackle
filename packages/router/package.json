{
  "name": "@crackle/router",
  "version": "0.2.1",
  "main": "dist/crackle-router.cjs.js",
  "module": "dist/crackle-router.esm.js",
  "exports": {
    ".": {
      "module": "./dist/crackle-router.esm.js",
      "default": "./dist/crackle-router.cjs.js"
    },
    "./routes": {
      "module": "./routes/dist/crackle-router-routes.esm.js",
      "default": "./routes/dist/crackle-router-routes.cjs.js"
    },
    "./server": {
      "module": "./server/dist/crackle-router-server.esm.js",
      "default": "./server/dist/crackle-router-server.cjs.js"
    },
    "./package.json": "./package.json"
  },
  "author": "SEEK",
  "license": "MIT",
  "files": [
    "/dist",
    "/routes",
    "/server",
    "valid-routes.d.ts"
  ],
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "routes.ts",
      "server.ts"
    ]
  },
  "dependencies": {
    "react": "^18.2.0",
    "react-router-dom": "6.4.3"
  },
  "devDependencies": {
    "@types/react": "^18.0.25"
  }
}
