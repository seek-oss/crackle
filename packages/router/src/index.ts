export type { RouteData } from './types';

// eslint-disable-next-line import-x/export
export { Link, crackleLink } from './Link';
export { createRouteData } from './create-route-data';

export type { ValidRoute } from './valid-routes';

// eslint-disable-next-line import-x/export
export * from 'react-router';
