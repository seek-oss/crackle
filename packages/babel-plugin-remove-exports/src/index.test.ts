import pluginTester from 'babel-plugin-tester/pure';

import plugin from './';

const tests = [
  {
    title: 'Used import',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      import { routeSetterUpper, unusedFunctions } from './internalRoutingLogic';

      import { Something } from '../Somewhere';

      export const routeData = createRouteData<BraidMetadata>(() => {
        const myRoute = '/details';
        return {
          route: myRoute,
          someDetail: routeSetterUpper(),
          globalMetadata: {
            isDeprecated: true,
            componentName: 'Details',
          },
        }
      });

      export default function () {
        return <Something />;
      }`,
  },
  {
    title: 'Braid import',
    code: `
      import { createRouteData } from '@crackle/router';
      import { Card, Stack } from 'braid-design-system';
      import React from 'react';

      import { routeSetterUpper, unusedFunctions } from './internalRoutingLogic';

      import { Something } from '../Somewhere';

      export const routeData = createRouteData<BraidMetadata>(() => {
        const myRoute = '/details';
        return {
          route: myRoute,
          someDetail: routeSetterUpper(),
          globalMetadata: {
            isDeprecated: true,
            componentName: 'Details',
          },
        }
      });

      export default function () {
        return (
          <Card>
            <Stack space="medium">
              <Something />
              <Something />
            </Stack>
          </Card>
        )
      }`,
  },
  {
    title: 'Used object',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      const myMetadata = {
        isDeprecated: true,
        componentName: 'Details',
      }

      export const routeData = createRouteData<BraidMetadata>(() => ({
        route: '/details',
        globalMetadata: myMetadata,
      }));

      const MyComponent = () => <div />

      export default function () {
        return <MyComponent />;
      }`,
  },
  {
    title: 'Nested object',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      const componentName = 'Details';

      const myMetadata = {
        isDeprecated: true,
        componentName,
      };

      export const routeData = createRouteData<BraidMetadata>(() => ({
        route: '/details',
        globalMetadata: myMetadata,
      }));

      const MyComponent = () => <div />;

      export default function () {
        return <MyComponent />;
      }`,
  },
  {
    title: 'Super nested object',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      const componentName = 'Details';

      const myMetadata = {
        isDeprecated: true,
        componentName,
      };

      const routeDetails = {
        route: '/details',
        globalMetadata: myMetadata,
      }

      export const routeData = createRouteData<BraidMetadata>(() => routeDetails);

      const MyComponent = () => <div />;

      export default function () {
        return <MyComponent />;
      }`,
  },
  {
    title: 'Separated named export',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      const routeData = createRouteData<BraidMetadata>(() => ({
        route: '/details',
        globalMetadata: {
          isDeprecated: true,
          componentName: 'Details',
        },
      }));

      const MyComponent = () => <div />;

      export default function () {
        return <MyComponent />;
      }

      export { routeData };`,
  },
  {
    title: 'Renamed export',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      const routes = createRouteData<BraidMetadata>(() => ({
        route: '/details',
        globalMetadata: {
          isDeprecated: true,
          componentName: 'Details',
        },
      }));

      const MyComponent = () => <div />;

      export default function () {
        return <MyComponent />;
      }

      export { routes as routeData };`,
  },
  {
    title: 'Local variable used in an imported function',
    code: `
      import { createRouteData } from '@crackle/router';
      import React from 'react';

      import { routeSetupFunc } from './settyThing';

      const componentDetails = 'siteEn';

      const routeData = createRouteData<BraidMetadata>(() => ({
        route: '/details',
        globalMetadata: {
          detailsKey: routeSetupFunc(componentDetails),
          isDeprecated: true,
          componentName: 'Details',
        },
      }));

      const MyComponent = () => <div />;

      export default function () {
        return <MyComponent />;
      }

      export { routeData };
`,
  },
];

pluginTester({
  pluginName: 'babel-plugin-remove-exports',
  plugin,
  babelOptions: {
    filename: 'test-file.tsx',
    plugins: [
      '@babel/plugin-syntax-jsx',
      ['@babel/plugin-syntax-typescript', { isTSX: true }],
    ],
  },
  snapshot: true,
  tests,
});
