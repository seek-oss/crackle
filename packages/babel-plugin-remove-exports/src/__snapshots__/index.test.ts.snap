// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`babel-plugin-remove-exports > 1. Braid import > 1. Braid import 1`] = `
import { createRouteData } from "@crackle/router";
import { Card, Stack } from "braid-design-system";
import React from "react";

import { routeSetterUpper, unusedFunctions } from "./internalRoutingLogic";

import { Something } from "../Somewhere";

export const routeData = createRouteData<BraidMetadata>(() => {
  const myRoute = "/details";
  return {
    route: myRoute,
    someDetail: routeSetterUpper(),
    globalMetadata: {
      isDeprecated: true,
      componentName: "Details",
    },
  };
});

export default function () {
  return (
    <Card>
      <Stack space="medium">
        <Something />
        <Something />
      </Stack>
    </Card>
  );
}
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
import { routeSetterUpper, unusedFunctions } from "./internalRoutingLogic";
export const routeData = createRouteData<BraidMetadata>(() => {
  const myRoute = "/details";
  return {
    route: myRoute,
    someDetail: routeSetterUpper(),
    globalMetadata: {
      isDeprecated: true,
      componentName: "Details"
    }
  };
});
`;

exports[`babel-plugin-remove-exports > 2. Keep default export, Local variable used in an imported function > 2. Keep default export, Local variable used in an imported function 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

import { routeSetupFunc } from "./settyThing";

const componentDetails = "siteEn";

const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    detailsKey: routeSetupFunc(componentDetails),
    isDeprecated: true,
    componentName: "Details",
  },
}));

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}

export { routeData };
    

      ↓ ↓ ↓ ↓ ↓ ↓

import React from "react";
const MyComponent = () => <div />;
export default function () {
  return <MyComponent />;
}
`;

exports[`babel-plugin-remove-exports > 3. Local variable used in an imported function > 3. Local variable used in an imported function 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

import { routeSetupFunc } from "./settyThing";

const componentDetails = "siteEn";

const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    detailsKey: routeSetupFunc(componentDetails),
    isDeprecated: true,
    componentName: "Details",
  },
}));

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}

export { routeData };
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
import { routeSetupFunc } from "./settyThing";
const componentDetails = "siteEn";
const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    detailsKey: routeSetupFunc(componentDetails),
    isDeprecated: true,
    componentName: "Details"
  }
}));
export { routeData };
`;

exports[`babel-plugin-remove-exports > 4. Nested object > 4. Nested object 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

const componentName = "Details";

const myMetadata = {
  isDeprecated: true,
  componentName,
};

export const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: myMetadata,
}));

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
const componentName = "Details";
const myMetadata = {
  isDeprecated: true,
  componentName
};
export const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: myMetadata
}));
`;

exports[`babel-plugin-remove-exports > 5. Renamed export > 5. Renamed export 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

const routes = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    isDeprecated: true,
    componentName: "Details",
  },
}));

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}

export { routes as routeData };
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
const routes = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    isDeprecated: true,
    componentName: "Details"
  }
}));
export { routes as routeData };
`;

exports[`babel-plugin-remove-exports > 6. Separated named export > 6. Separated named export 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    isDeprecated: true,
    componentName: "Details",
  },
}));

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}

export { routeData };
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: {
    isDeprecated: true,
    componentName: "Details"
  }
}));
export { routeData };
`;

exports[`babel-plugin-remove-exports > 7. Super nested object > 7. Super nested object 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

const componentName = "Details";

const myMetadata = {
  isDeprecated: true,
  componentName,
};

const routeDetails = {
  route: "/details",
  globalMetadata: myMetadata,
};

export const routeData = createRouteData<BraidMetadata>(() => routeDetails);

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
const componentName = "Details";
const myMetadata = {
  isDeprecated: true,
  componentName
};
const routeDetails = {
  route: "/details",
  globalMetadata: myMetadata
};
export const routeData = createRouteData<BraidMetadata>(() => routeDetails);
`;

exports[`babel-plugin-remove-exports > 8. Used import > 8. Used import 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

import { routeSetterUpper, unusedFunctions } from "./internalRoutingLogic";

import { Something } from "../Somewhere";

export const routeData = createRouteData<BraidMetadata>(() => {
  const myRoute = "/details";
  return {
    route: myRoute,
    someDetail: routeSetterUpper(),
    globalMetadata: {
      isDeprecated: true,
      componentName: "Details",
    },
  };
});

export default function () {
  return <Something />;
}
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
import { routeSetterUpper, unusedFunctions } from "./internalRoutingLogic";
export const routeData = createRouteData<BraidMetadata>(() => {
  const myRoute = "/details";
  return {
    route: myRoute,
    someDetail: routeSetterUpper(),
    globalMetadata: {
      isDeprecated: true,
      componentName: "Details"
    }
  };
});
`;

exports[`babel-plugin-remove-exports > 9. Used object > 9. Used object 1`] = `
import { createRouteData } from "@crackle/router";
import React from "react";

const myMetadata = {
  isDeprecated: true,
  componentName: "Details",
};

export const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: myMetadata,
}));

const MyComponent = () => <div />;

export default function () {
  return <MyComponent />;
}
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { createRouteData } from "@crackle/router";
const myMetadata = {
  isDeprecated: true,
  componentName: "Details"
};
export const routeData = createRouteData<BraidMetadata>(() => ({
  route: "/details",
  globalMetadata: myMetadata
}));
`;
